wsclient/src/main/java/org/hpccsystems/ws/client/HPCCWsWorkUnitsClient.java

```diff
@@ -378,10 +379,15 @@ static public boolean isWorkunitComplete(WUState state)
      */
     public static WUState getStateID(WorkunitInfo wu)
     {
-        if (wu != null)
-            return getStateID(wu.getStateID());
-        else
-            return WUState.UNKNOWN;
+        if (wu != null) {
+            if (wu.getStateID() != null) {
+                return getStateID(wu.getStateID());
+            } else if (wu.getState() != null && wu.getState().trim().length() != 0) {
```

Comment:
we should either trim in translateWUState() or make sure the argument in the translateWUstate call below is trimmed as well. As is, we might pass this condition (wu.getState().trim().length() != 0), but fail to translate due to leading or trailing spaces on an otherwise valid state string.